prophy/generators.py

```diff
@@ -29,6 +29,69 @@ def _build_up_implementation(cls):
         """
             Implementation of type creation. To be overridden in derived metaclasses.
         """
+    def __eq__(cls, other):
+        if not isinstance(other, cls.__class__):
+            return NotImplemented
+
+        if cls.__bases__ != other.__bases__:
+            return False
+
+        for cls_d, other_d in zip(cls._descriptor, other._descriptor):
+            name_eq = cls_d.name == other_d.name
+            if not name_eq:
+                return False
+
+            if cls_d.discriminator != other_d.discriminator:
+                return False
+
+            if cls_d.type.__name__ == other_d.type.__name__ == '_array':
```

Comment:
Since you call `cls_d.type` and `other_d.type` dozen times, please make a helper like that and use whenever it's possible:
```python
    def are_eq(attribute_name):
        if hasattr(cls_d.type, attribute_name) and hasattr(other_d.type, attribute_name):
            return getattr(cls_d.type, attribute_name) == getattr(other_d.type, attribute_name)

    def are_not_eq(attribute_name):
        return not are_eq(attribute_name)
```
then the code from lines L69-75 would become:
```python
    if are_not_eq("__bases__"):
        return False

    for f in ('_ALIGNMENT', '_BOUND', '_BOUND_SHIFT', '_DEFAULT', '_DYNAMIC', '_OPTIONAL', '_SIZE',
              '_UNLIMITED'):
        if are_not_eq(f):
            return False

```